keps/sig-cli/kubectl-unix-socket.md

```diff
@@ -0,0 +1,177 @@
+    - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+    - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+- [Implementation History](#implementation-history)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location
+in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Supporting unix socket for `kubectl` is a new feature to provide non-stored credentials for kubeconfig. 
+
+## Motivation
+
+Currently kubectl does not expose a way to send the request over a unix socket. `Curl` does provide a `--unix-socket` option,
+as do many other cli utilities.
+
+Currently several kinds of credentials (passwords, tokens, x509 certificates and etc) are stored in a kubeconfig file and
+persisted on disk. It is not safe and controllable, especially this kubeconfig gets spread to everywhere. Moreover,
+it will become difficult to revoke the access to the apiserver.
+
+Unix sockets are used as a secured proxy into some infrastructure and it would be great to be able to send requests
+from kubectl over this unix socket.
+
+### Goals
+
+* kubectl users can use unix socket to connect to the apiserver:
+    * Add an experimental flag `--experimental-unix-socket` for kubectl, much like `--unix-socket` in `curl`;
+    * Allow specifying unix sockets in kubeconfig file;
+
+### Non-Goals
+
+* Convey credentials through unix sockets;
+* Adopting [SGX] to secure the memory access;
+
+[SGX]: https://www.intel.com/content/www/us/en/architecture-and-technology/software-guard-extensions.html
+
+## Proposal
+
+### User Stories
+
+Issuing commands to any kube api server whereby unix socket is the preferred way to access it.
+
+```bash
+kubectl --experimental-unix-socket ~/.myunixsocket --server mykubecluster.com get pods
+```
+
+### Implementation Details/Notes/Constraints
+
+A `--unix-socket` flag will be added to `kubectl`, much like the --unix-socket in `curl`. The curl docs state: (HTTP)
```

Comment:
I was expecting this to look more like allowing `unix://` as a protocol in the existing server field and command-line flags (where we currently allow http and https)

Reply:
Yes, this will be better.